### PR TITLE
Error logging and user attributes

### PR DESCRIPTION
**Pull request by JonnyLinja**
Moved the mParticle specific stuff to the subspec folder

### CODE REVIEW

**Comment by JonnyLinja** [comment]
Note that I also removed the default Adobe implementations here, which was already done separately by CJ already.

File: Simcoe/Simcoe.swift [hunk 1]
```diff
@@ -110,26 +110,6 @@ public final class Simcoe {
...
         engine.trackLocation(location, withAdditionalProperties: properties)
     }
-    
-    /**
-     Logs the error with optional additional properties.
-     
-     - parameter error:      The error to log.
-     - parameter properties: The optional additional properties.
-     */
-    public static func logError(error: String, withAdditionalProperties properties: Properties? = nil) {
```
Comment (MichaelCampbell) [inline Simcoe/Simcoe.swift:120]: i think we should leave this here, as it's generic enough and will likely be reused with other analytics providers

File: Simcoe/Simcoe.swift [hunk 2]
```diff
@@ -110,26 +110,6 @@ public final class Simcoe {
...
-        engine.logError(error, withAdditionalProperties: properties)
-    }
-    
-    /**
-     Sets the User Attribute.
-     
-     - parameter key:   The key of the user attribute
-     - parameter value: the value of the user attribute
-     */
-    public static func setUserAttribute(key: String, value: AnyObject) {
```
Comment (MichaelCampbell) [inline Simcoe/Simcoe.swift:130]: same here. Both mParticle and Segment use a similar structure

**Comment by MichaelCampbell** [comment]
Hey @JonnyLinja any chance you will have time to work on this? If not please close, and I will recreate with the necessary updates. Let me know either way

**Pull request closed without merging**